extensions-core/hdfs-storage/pom.xml

```diff
@@ -355,6 +355,16 @@
         <artifactId>hadoop-hdfs-client</artifactId>
         <scope>runtime</scope>
       </dependency>
+      <dependency>
```

Comment:
Please include a comment about why these are necessary (they aren't directly used, nor are they pulled in transitively [apparently], which makes them mysterious and in need of a comment).

Reply:
Added a comment

Reply:
If the dependency analyze check complains, you can change the `scope` to `runtime` for these.